# Fantasy Book Analysis

## Introduction

The goal of this project is to analyze data from Goodreads (https://www.goodreads.com/), with a focus on books of the Fantasy genre. I will aim to retrieve insights such as:
- Are specific sub-genres (epic, dark, romance) rated higher?
- Does publication year affect ratings?
- Do longer fantasy books get higher ratings?
- How did the ratings/views change over the last few years?

### Data

The data used in this exercise will be downloaded from Kaggle (https://www.kaggle.com/datasets/middlelight/goodreadsbookswithgenres). It includes information about genre, publisher, average review, etc. Let's take a look at the data.

In [23]:
# install libraries
#import sys
#!"{sys.executable}" -m pip install -U pip
#!"{sys.executable}" -m pip install pandas numpy seaborn matplotlib

# import libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

# read in data
books = pd.read_csv("data/Goodreads_books_with_genres.csv")

# how many rows and cols
books.shape

(11127, 13)

In [24]:
# view the first 5 rows
books.head()

Book Id                                              Title  \
0        1  Harry Potter and the Half-Blood Prince (Harry ...   
1        2  Harry Potter and the Order of the Phoenix (Har...   
2        4  Harry Potter and the Chamber of Secrets (Harry...   
3        5  Harry Potter and the Prisoner of Azkaban (Harr...   
4        8  Harry Potter Boxed Set  Books 1-5 (Harry Potte...   

                       Author  average_rating        isbn         isbn13  \
0  J.K. Rowling/Mary GrandPré            4.57  0439785960  9780439785969   
1  J.K. Rowling/Mary GrandPré            4.49  0439358078  9780439358071   
2                J.K. Rowling            4.42  0439554896  9780439554893   
3  J.K. Rowling/Mary GrandPré            4.56  043965548X  9780439655484   
4  J.K. Rowling/Mary GrandPré            4.78  0439682584  9780439682589   

  language_code  num_pages  ratings_count  text_reviews_count  \
0           eng        652        2095690               27591   
1           eng        870        2153167               29221   
2           eng        352           6333                 244   
3           eng        435        2339585               36325   
4           eng       2690          41428                 164   

  publication_date        publisher  \
0        9/16/2006  Scholastic Inc.   
1         9/1/2004  Scholastic Inc.   
2        11/1/2003       Scholastic   
3         5/1/2004  Scholastic Inc.   
4        9/13/2004       Scholastic   

                                              genres  
0  Fantasy;Young Adult;Fiction;Fantasy,Magic;Chil...  
1  Fantasy;Young Adult;Fiction;Fantasy,Magic;Chil...  
2  Fantasy;Fiction;Young Adult;Fantasy,Magic;Chil...  
3  Fantasy;Fiction;Young Adult;Fantasy,Magic;Chil...  
4  Fantasy;Young Adult;Fiction;Fantasy,Magic;Adve...

We have 11,127 rows and 13 columns. The columns include information about the publication year, authors, ISBN, genres, and other important aspects that we will explore and use.